### PR TITLE
First pass of definition of programming language

### PR DESCRIPTION
**Pull request by zspencer**

### CODE REVIEW

**Comment by zspencer** [comment]
Feedback welcome @openspectrum

File: _posts/2014-01-01-programming-language.md [hunk 1]
```diff
@@ -0,0 +1,27 @@
+A programming language bridges the gap between [machine code](/machine-code)
+and human language. It's a tool for humans to provide
+[instructions](/instruction) a computer can understand. The fist programming
```
Comment (tannerwelsh) [inline _posts/2014-01-01-programming-language.md]: *first

Comment (tannerwelsh) [inline _posts/2014-01-01-programming-language.md]: Also this sentence probably belongs in a separate paragraph, as doesn't fit well with the other introductory sentences.

File: _posts/2014-01-01-programming-language.md [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+languages have rules for how to structure code, called [syntax](/syntax). When
+these rules are broken, the programming language will provide a [syntax
+error](/syntax#syntax-error).
+
+Some programming languages, like Ruby, try to avoid forcing people to conform
+to the machine's pedanticism. While this seems like a good idea (Yay! Fewer
+parenthesis!) it may result in ambiguous code which looks valid but is not.
+
+Programming languages are often grouped based on whether they:
+* Must be [compiled](/compiler) before being ran, or whether they may be
```
Comment (tannerwelsh) [inline _posts/2014-01-01-programming-language.md]: s/ran/run

File: _posts/2014-01-01-programming-language.md [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+  [interpretted](/interpreter) on the fly
+* Validate data [conforms to a declared type](/strong-typing) or if they allows
```
Comment (tannerwelsh) [inline _posts/2014-01-01-programming-language.md]: *allow

File: _posts/2014-01-01-programming-language.md [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+  [types to be modified at runtime](/weak-typing).
+
+Most programming languages have two ways to interact with them. You may:
```
Comment (tannerwelsh) [inline _posts/2014-01-01-programming-language.md]: Shouldn't this be "Most _interpreted_ programming languages..." ?

**Comment by tannerwelsh** [comment]
Overall this is good. Maybe provide a small list of example languages that fall into each category? Probably: C (compiled, strong), Clojure (compiled, weak), Ruby (interpreted, weak), not many common examples of interpreted / strong typed.

**Comment by zspencer** [comment]
Decided to clean up these outstanding PRs

**Pull request merged**